### ISSUE TITLE
Desktop > Wallpaper: Missing button of added image on subsequent visits to the page

### ISSUE THREAD

**Issue by git-f0x** [pop-os/cosmic-settings#252]
When the Wallpaper page is first opened, the added image is present both on the list and in the large preview.
When going to another page and returning (without closing Settings), the added image is missing in the list ~~and the preview is black~~ (though both appear for a split second), until scrolling or something is hovered over.

https://github.com/pop-os/cosmic-settings/assets/150025636/3d6582c9-a5d9-43c3-9416-d587ababcd12

**Comment by git-f0x** [comment]
The preview being blank is fixed, but the added wallpaper still doesn't show up on the list until something is interacted with on subsequent visits to the page.

**Comment by joshuamegnauth54** [comment]
Sorry! I'm responding to all of this a bit late. :laughing: 

I attempted to fix this issue over the last few days but I haven't made much progress. The PR you commented on fixed a related issue (which was likely a regression I caused!) but doesn't do much for this issue.

I'll likely continue trying to fix this issue, but I'm definitely stumped for now. The custom wallpaper buttons are successfully created despite not appearing. It's weird.
